src/adapters/chineseloc/adapter.js

```diff
@@ -0,0 +1,144 @@
+
+      lexemes.push(lexmodel)
+    })
+
+    let finalLexemes = []
+    lexemes.forEach(lex => {
+      let check = finalLexemes.filter(checkLex => {
+        let check1 = checkLex.lemma.principalParts[0] === lex.lemma.principalParts[0]
+
+        let check2 = checkLex.lemma.features[Feature.types.pronunciation].value === lex.lemma.features[Feature.types.pronunciation].value
+
+        let check3 = checkLex.meaning.shortDefs[0].text === lex.meaning.shortDefs[0].text
+
+        return check1 && check2 && check3
+      })
+
+      if (check.length === 0) {
+        finalLexemes.push(lex)
+      }
+    })
+
+    if (finalLexemes.length > 0) {
+      return new Homonym(finalLexemes, targetWord)
+    } else {
+      return undefined
+    }
+  }
+
+  extractFeatures (rawLexeme) {
+    let featuresArr = [
+      { checkAttribute: 'pinyin', method: this.defineMultipleFeature.bind(this), featureType: Feature.types.pronunciation, featOrder: 4 },
+      { checkAttribute: 'format', method: this.defineFeatureFormat.bind(this), featureType: Feature.types.note },
+      { checkAttribute: 'mandarin', method: this.defineMultipleFeature.bind(this), featureType: Feature.types.pronunciation, featOrder: 3 },
+      { checkAttribute: 'cantonese', method: this.defineMultipleFeature.bind(this), featureType: Feature.types.pronunciation, featOrder: 2 },
+      { checkAttribute: 'tang', method: this.defineMultipleFeature.bind(this), featureType: Feature.types.pronunciation, featOrder: 1 },
+      { checkAttribute: 'frequency', method: this.defineSimpleFeature.bind(this), featureType: Feature.types.frequency },
+      { checkAttribute: 'unicode', method: this.defineSimpleFeature.bind(this), featureType: Feature.types.radical }
+    ]
+    let features = []
+
+    featuresArr.forEach(featureConfig => {
+      let featureVal = featureConfig.method(featureConfig, rawLexeme, features)
+      if (featureVal) {
+        features.push(featureVal)
+      }
+    })
+    return features
+  }
+
+  defineMultipleFeature (featureConfig, rawLexeme, features) {
+    if (!rawLexeme[featureConfig.checkAttribute]) {
+      return
+    }
+
+    let featType = featureConfig.featureType
+    let featObj = features.filter(feat => feat.type === featType)
+
+    if (featObj.length === 0) {
+      return new Feature(featureConfig.featureType, [[rawLexeme[featureConfig.checkAttribute], featureConfig.featOrder]], this.languageID)
+    } else {
+      featObj[0].addValue(rawLexeme[featureConfig.checkAttribute], featureConfig.featOrder)
+    }
+  }
+
+  defineSimpleFeature (featureConfig, rawLexeme) {
+    if (rawLexeme[featureConfig.checkAttribute]) {
+      return new Feature(featureConfig.featureType, rawLexeme[featureConfig.checkAttribute], this.languageID)
+    }
+  }
+
+  defineFeatureFormat (featureConfig, rawLexeme) {
+    if (rawLexeme[featureConfig.checkAttribute]) {
+      let value = null
+      if (rawLexeme[featureConfig.checkAttribute] === 'simp') {
+        value = 'simple'

```

Comment:
yes agree

Reply:
(and also with the comment made in the data-models PR that really the right word is "simplified" not "simple")

Reply:
I don't agree. Similiar to the frequency feature.
We should decide where to format features.
All fetaures are recieved from the source and are not formatted in ClientAdapter for otheres source.
So I think that I should move this conversion to the chineseSource - chinese service implementation for now.
@balmas and @kirlat , what do you think?

Reply:
I agree with @irina060981 that it's better to do during conversion of Chinese data. In that case we can do it only once and avoid dynamic conversion every time request comes in. 

I also think that since an architecture of our Chinese adapter is in flux we can probably postpone it for later but add a TODO to remind us about this issue. I, however, think that we should not delay changing `simple` to `simplified`.

Reply:
Yes, I will fix format conversion and move it to chineseSource

Reply:
Fixed